### ISSUE TITLE
Stop reconnecting

### ISSUE THREAD

**Issue by skadyrov** [pusher/pusher-websocket-java#189]
### What is the issue?
I am using this library in an Android app. My desired behavior when the app goes into background:
1. Disconnect if already connected
2. Stop reconnecting if it was not connected yet.

For disconnecting I use method:
```
pusher.disconnect()
```
So far it works fine if the connection state is `ConnectionState.CONNECTED`. 

However, if the connection state was `ConnectionState.CONNECTING` or `ConnectionState.RECONNECTING`, using the `disconnect()` method doesn't stop the pusher from attempting to connect. I also did not find any other way to stop attempting to reconnect without having to implement a workaround. It would be nice to have a functionality to do both of the steps I described directly with pusher.

### Is it a crash report? 
 - No

### Any improvements you suggest

...

---- 

CC @pusher/mobile

**Comment by jmanini** [comment]
I agree it would be helpful to stop reconnecting attempts. My workaround has been to check for spurious connection when ConnectionState.CONNECTED and then invoke pusher.disconnect().

**Comment by skadyrov** [comment]
@jmanini I used the same workaround :)

**Comment by petarmarijanovicfive** [comment]
Still no improvement here? 😁

**Comment by Vlad-M-1** [comment]
It seems that it still isn't implemented.
How to stop current connection attempt?

According to Pusher.java:
```

    public void disconnect() {
        if (connection.getState() == ConnectionState.CONNECTED) {
            connection.disconnect();
        }
    }

```
Moreover, according to WebSocketConnection.java:

```
    @Override
    public void disconnect() {
        factory.queueOnEventThread(new Runnable() {
            @Override
            public void run() {
                if (state == ConnectionState.CONNECTED) {
                    updateState(ConnectionState.DISCONNECTING);
                    underlyingConnection.close();
                }
            }
        });
    }
```

So basically there's a lack of opportunities to stop connection when conenctionState is other than ConnectionState.CONNECTED.
